CantoboardFramework/Keyboard/View/KeyboardView.swift

```diff
@@ -279,34 +279,27 @@ class KeyboardView: UIView, InputView {
             // let rimeSchema = state.rimeSchema
             keyCaps = keyCaps.map { $0.map {
                 switch $0 {
-                case .character(let c) where state.activeSchema.isCangjieFamily && c.first?.isEnglishLetter ?? false && state.inputMode != .english:
-                    return .cangjie(c)
-                case .character("F"), .character("G"), .character("H"),
-                     .character("C"), .character("V"), .character("B"),
-                     .character("f"), .character("g"), .character("h"),
-                     .character("c"), .character("v"), .character("b"):
-                    switch state.keyboardContextualType {
-                    case .rime, .url(true):
-                        if case .character(let c) = $0,
-                           state.activeSchema == .jyutping && Settings.cached.toneInputMode == .longPress || state.activeSchema == .yale {
-                            // Show tone keys.
-                            return .characterWithConditioanlPopup(c)
-                        } else {
-                            return $0
+                case .character(let c) where state.inputMode != .english:
```

Comment:
Your code is much cleaner, thanks.

Reply:
This is changed because the tone hints should be hidden in English input mode.